### PR TITLE
ci: self-service compat test start

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
It should now be possible for anyone to trigger the daily compat build by adding a `run-full-compat: true` trailer to their commit message.

```
CHANGELOG_BEGIN
CHANGELOG_END
```

run-full-compat: true

### CODE REVIEW

**Comment by tudor-da** [comment]
Good one!

File: ci/prs.yml [hunk 1]
```diff
@@ -170,3 +170,22 @@ jobs:
...
+    vmImage: ubuntu-20.04
+  steps:
+  - checkout: self
+  - template: bash-lib.yml
+    parameters:
+      var_name: bash-lib
+  - bash: |
+      set -euo pipefail
+      source $(bash-lib)
+      COMMIT=$(git rev-parse HEAD^2)
```
Comment (tudor-da) [inline ci/prs.yml:185]: Out of curiosity, why `HEAD^2`?

**Review by tudor-da — APPROVED** [review]

File: ci/prs.yml (hunk 1 above)
Comment (garyverhaegen-da) [reply ci/prs.yml:185]: PR builds (so whatever's in `ci/prs.yml`) run on a synthesized merge commit between the branch and `main`, so `HEAD^1` is the main commit and `HEAD^2` is the branch commit (not to be confused with `HEAD~2`, which would be the parent of the main commit).

**Review by cocreature — COMMENTED** [review]
Nice, thank you!

**Pull request merged**